### PR TITLE
Initial Jest configuration

### PR DESCRIPTION
**Pull request by SamuelHarner**
Installed the initial tools I believe we will utilize for testing, this includes:
- Jest
- React test renderer
- React Native testing library

Also added some initial settings like jest-expo preset, and a trivial test (in file `index.test.js`) that can be run to check that Jest is working. (remember to run `npm install` to make sure you have the required dependencies)

Added a section to README for info about testing and how to run tests, testing guidelines/strategy will also be added to this section.

One thing I am unsure about is if Jest requires additional configuration to work properly with .jsx files. However, I think some simple tests could be written to test this now, for example test some aspect of index.jsx or authors.jsx.

### CODE REVIEW

**Comment by editf** [comment]
I haven't had the time to research this properly yet but based on [these jest docs](https://jestjs.io/docs/en/tutorial-react) I think we might need to add ```babel-jest``` in order to test .jsx files?

**Comment by sanlil** [comment]
Looks good to me! Would be great if you could add some simple logic and add a test for that to try it out.

File: package.json [hunk 1]
```diff
@@ -5,7 +5,19 @@
     "android": "expo start --android",
     "ios": "expo start --ios",
     "web": "expo start --web",
-    "eject": "expo eject"
+    "eject": "expo eject",
+    "test": "node_modules/.bin/jest"
+  },
+  "jest": {
+    "preset": "jest-expo",
+    "silent": true,
```
Comment (david8611) [inline package.json:13]: This will hide console.log errors.

File: App/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Authors from './screens/Authors';
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
+  ReactDOM.render(<Authors />, div);
+});
```
Comment (david8611) [inline App/index.test.jsx:12]: Added a simple render test, can be found here: https://create-react-app.dev/docs/running-tests/

**Comment by david8611** [comment]
> I haven't had the time to research this properly yet but based on [these jest docs](https://jestjs.io/docs/en/tutorial-react) I think we might need to add `babel-jest` in order to test .jsx files?

I found this to be correct and therefore added support for ```babel-jest``` which enables imports and other functionality to be able to test components, etc.

**Review by david8611 — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -22,6 +27,8 @@
     "@react-navigation/material-bottom-tabs": "^5.3.13",
     "@react-navigation/native": "^5.9.2",
     "@react-navigation/stack": "^5.14.2",
+    "@testing-library/react": "^11.2.5",
+    "babel-jest": "^26.6.3",
```
Comment (SamuelHarner) [inline package.json:31]: Maybe these should be devDependencies instead?

Some sources I found:
- https://stackoverflow.com/questions/56705177/should-jest-enzyme-be-a-devdependency-or-dependency
- https://github.com/slackhq/csp-html-webpack-plugin/issues/32

File: package.json [hunk 4]
```diff
@@ -51,7 +58,9 @@
     "jest": "^26.6.3",
     "jest-expo": "^40.0.2",
     "prettier": "2.2.1",
-    "react-test-renderer": "^17.0.1"
+    "react-test-renderer": "^17.0.1",
+    "ts-jest": "^26.5.1",
+    "typescript": "^4.1.4"
```
Comment (SamuelHarner) [inline package.json:64]: Are these necessary if we are not using Typescript?

File: package.json (hunk 3 above)
Comment (david8611) [reply package.json:31]: You're probably right. I will adjust this in a seperate PR.

File: package.json (hunk 4 above)
Comment (david8611) [reply package.json:64]: No, they are not necessary. I will remove them in a separate PR.

**Pull request merged**